perllib/FixMyStreet/Roles/ParallelAPI.pm

```diff
@@ -35,7 +42,10 @@ sub call_api {
     } else {
         # uncoverable statement
         system(@cmd);
-        $data = Storable::fd_retrieve($tmp);
+        unless ($?) {
+            $data = Storable::retrieve($tmp);
+            unlink $tmp; # don't want to inadvertently cache forever
+        }
     }
     $c->session->{$key} = $data;
```

Comment:
Presumably don't want to do this (or the logging the call time) if there was no data?